server/src/FeatureFinder.ts

```diff
@@ -50,66 +32,69 @@ export class FeatureFinder {
+  }
+
+  /** Tries to find a suitable command to start the server first in the current Python environment and then in the global scope.
+   * Installs dbt-rpc for dbt version > 1.0.0.
+   * @returns {Command} or `undefined` if nothing is found
+   */
+  async findDbtRpcCommand(dbtProfileType?: string): Promise<Command | undefined> {
+    const [dbtRpcPythonVersion, dbtPythonVersion, dbtRpcGlobalVersion] = await this.availableCommandsPromise;
 
-    if (dbtRpcPythonVersion?.installedVersion && dbtRpcPythonVersion.installedAdapter) {
+    if (dbtRpcPythonVersion?.installedVersion && dbtRpcPythonVersion.installedAdapters.some(a => a.name === dbtProfileType)) {
       this.versionInfo = dbtRpcPythonVersion;
-      return new DbtRpcCommand(FeatureFinder.DBT_RPC_PARAMS, this.pythonInfo?.path);
+      return this.dbtCommandFactory.getDbtRpcRun();
     }
     if (dbtPythonVersion?.installedVersion) {
       this.versionInfo = dbtPythonVersion;
       return dbtPythonVersion.installedVersion.major >= 1
         ? this.installAndFindCommandForV1(dbtProfileType)
-        : new DbtCommand(FeatureFinder.LEGACY_DBT_PARAMS, this.pythonInfo?.path);
+        : this.dbtCommandFactory.getLegacyDbtRpcRun();
     }
-    if (dbtRpcGlobalVersion?.installedVersion && dbtRpcGlobalVersion.installedAdapter) {
+    if (dbtRpcGlobalVersion?.installedVersion && dbtRpcGlobalVersion.installedAdapters.some(a => a.name === dbtProfileType)) {
       this.versionInfo = dbtRpcGlobalVersion;
-      return new DbtRpcCommand(FeatureFinder.DBT_RPC_PARAMS);
+      return this.dbtCommandFactory.getGlobalDbtRpcRun();
     }
 
     return undefined;
   }
 
-  async findGlobalDbtCommand(dbtProfileType?: string): Promise<boolean> {
-    const settledResults = await Promise.allSettled([this.findDbtPythonVersion(dbtProfileType), this.findDbtGlobalVersion(dbtProfileType)]);
+  async findPythonForCli(dbtProfileType?: string): Promise<string | undefined> {
+    const [, dbtPythonVersion, , dbtGlobalVersion] = await this.availableCommandsPromise;
 
-    const [dbtPythonVersion, dbtGlobalVersion] = settledResults.map(v => {
-      return v.status === 'fulfilled' ? v.value : undefined;
-    });
-
-    let versions = '';
-    versions += dbtPythonVersion ? `dbtPythonVersion = ${getStringVersion(dbtPythonVersion.installedVersion)} ` : '';
-    versions += dbtGlobalVersion ? `dbtGlobalVersion = ${getStringVersion(dbtGlobalVersion.installedVersion)}` : '';
-
-    console.log(versions);
-
-    if (dbtPythonVersion?.installedVersion) {
+    if (dbtPythonVersion?.installedVersion && dbtPythonVersion.installedAdapters.some(a => a.name === dbtProfileType)) {
       this.versionInfo = dbtPythonVersion;
-    } else if (dbtGlobalVersion?.installedVersion) {
+      return this.getPythonPath();
+    }
+
+    if (dbtGlobalVersion?.installedVersion && dbtGlobalVersion.installedAdapters.some(a => a.name === dbtProfileType)) {
```

Comment:
How it is related to `findPythonForCli` ? Please think about naming.

Reply:
renamed